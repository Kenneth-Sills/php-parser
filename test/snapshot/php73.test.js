const parser = require('../main');

describe('Test syntax parsing with PHP 73 support', function() {

  it('https://wiki.php.net/rfc/list_reference_assignment', function() {
    var ast = parser.parseEval('[$a, &$b] = $array; list($a, &$b) = $array;', {
      parser: {
        php7: true,
        php74: false
      }
    });
    expect(ast).toMatchSnapshot();
  });
});
