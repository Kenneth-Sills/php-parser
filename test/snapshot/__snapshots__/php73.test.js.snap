// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test syntax parsing with PHP 73 support https://wiki.php.net/rfc/list_reference_assignment 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": List {
          "items": Array [
            Variable {
              "curly": false,
              "kind": "variable",
              "name": "a",
            },
            ByRef {
              "kind": "byref",
              "what": Variable {
                "curly": false,
                "kind": "variable",
                "name": "b",
              },
            },
          ],
          "kind": "list",
          "shortForm": true,
        },
        "operator": "=",
        "right": Variable {
          "curly": false,
          "kind": "variable",
          "name": "array",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": List {
          "items": Array [
            Variable {
              "curly": false,
              "kind": "variable",
              "name": "a",
            },
            ByRef {
              "kind": "byref",
              "what": Variable {
                "curly": false,
                "kind": "variable",
                "name": "b",
              },
            },
          ],
          "kind": "list",
          "shortForm": false,
        },
        "operator": "=",
        "right": Variable {
          "curly": false,
          "kind": "variable",
          "name": "array",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
